iphone/Classes/TiUILabel.m

```diff
@@ -225,96 +225,77 @@ -(void)handleListenerRemovedWithEvent:(NSString *)event
-            }
-        }
+    // ask the layoutManager which glyph is under this tapped point
+    NSUInteger glyphIdx = [layoutManager glyphIndexForPoint:point
+                                            inTextContainer:textContainer
+                             fractionOfDistanceThroughGlyph:NULL];
+    
+    // as explained in Apple's documentation the previous method returns the nearest glyph
+    // if no glyph was present at that point. So if we want to ensure the point actually
+    // lies on that glyph, we should check that explicitly
+    CGRect glyphRect = [layoutManager boundingRectForGlyphRange:NSMakeRange(glyphIdx, 1)
+                                                inTextContainer:textContainer];
+    if (CGRectContainsPoint(glyphRect, point))
+    {
+        return [layoutManager characterIndexForGlyphAtIndex:glyphIdx];
+    } else {
+        return NSNotFound;
+    }
+}
+
+-(BOOL)checkLinkAttributeForString:(NSMutableAttributedString*)theString atPoint:(CGPoint)p
+{
+    NSUInteger idx = [self characterIndexAtPoint:theString atPoint:p];
+    if (idx == NSNotFound) {
+        return NO;
     }
     
-    //Don't need frame,path or lines now
-    CFRelease(frame);
-    CFRelease(path);
-    //CFRelease(lines);
     
-    if (idx != NSNotFound) {
-        if(idx > theString.string.length) {
-            return NO;
-        }
-        NSRange theRange = NSMakeRange(0, 0);
-        NSString *url = nil;
+    NSRange theRange = NSMakeRange(0, 0);
+    NSString *url = nil;
+    NSString *url2 = nil;
 #ifdef USE_TI_UIATTRIBUTEDSTRING
-        TiUIAttributedStringProxy *tempString = [[self proxy] valueForKey:@"attributedString"];
-        url = [tempString getLink:idx];
+    TiUIAttributedStringProxy *tempString = [[self proxy] valueForKey:@"attributedString"];
+    url = [tempString getLink:idx];
 #endif
-        if(url != nil && url.length) {
-            NSDictionary *eventDict = [NSDictionary dictionaryWithObjectsAndKeys:
-                                       url, @"url",
-                                       [NSArray arrayWithObjects:NUMUINTEGER(theRange.location), NUMUINTEGER(theRange.length),nil],@"range",
-                                       nil];
-                                            
-            [[self proxy] fireEvent:@"link" withObject:eventDict propagate:NO reportSuccess:NO errorCode:0 message:nil];
-            return YES;
-        }
+    
+    if (url == nil || url.length == 0) {
+        return NO;
     }
-    return NO;
+        
+    NSDictionary *eventDict = [NSDictionary dictionaryWithObjectsAndKeys:
+                                url, @"url",
+                                [NSArray arrayWithObjects:NUMUINTEGER(theRange.location), NUMUINTEGER(theRange.length),nil],@"range",
+                                    nil];
+    [[self proxy] fireEvent:@"link" withObject:eventDict propagate:NO reportSuccess:NO errorCode:0 message:nil];
```

Comment:
Add a check to see if the event-listener is added, e.g. `[[self proxy] _hasListeners:@"link"]`. It may be checked even before to prevent unnecessary memory usage. Maybe in the method call inside `recognizedTap:`. 

And finally, I noticed the method returns a bool but it's callee does not use the return-value. I never worked on that API before, can you think of we that was done?